public/app/core/components/RolePicker/UserRolePicker.tsx

```diff
@@ -63,14 +70,14 @@ export const UserRolePicker = ({
       console.error('Error loading options');
     }
     return [];
-  }, [orgId, userId, pendingRoles]);
+  }, [orgId, userId, pendingRoles, roles]);
 
   useEffect(() => {
     // only load roles when there is an Org selected
     if (orgId) {
       getUserRoles();
     }
-  }, [orgId, getUserRoles, pendingRoles]);
+  }, [orgId, getUserRoles, pendingRoles, roles]);
```

Comment:
Do we need to track `pendingRoles` and `roles` here? They are already tracked in the useAsyncFn dependency list and will be updated there when needed.